sound/soc/intel/boards/sdw_rt711_rt1308_rt715.c

```diff
@@ -770,21 +812,112 @@ static int create_sdw_codec_dai(struct device *dev,
 		if (!name)
 			return -ENOMEM;
 
-		cpu_name = devm_kasprintf(dev, GFP_KERNEL,
-					  "SDW%d Pin%d", link_id, j + 2);
-		if (!cpu_name)
-			return -ENOMEM;
+		for (k = 0; k < cpu_num; k++) {
+			int index = SDW_DISCO_LINK_ID(link->adr[k]);
 
-		init_dai_link(dai_links, *id, name, 1 - i, i, cpus, cpu_name,
+			cpu_name = devm_kasprintf(dev, GFP_KERNEL,
+						  "SDW%d Pin%d", index, j + 2);
+			if (!cpu_name)
+				return -ENOMEM;
+
+			cpu_dai[k].dai_name = cpu_name;
 			      codec, link->num_adr, *id, NULL, &sdw_ops);
+
+		init_dai_link(dai_links, *id, name, 1 - i, i, cpu_dai, cpu_num,
 		set_codec_init_func(link, dai_links + *id, 1 - i);
+		cpu_dai += cpu_num;
 		(*id)++;
 		j++;
 	}
 
 	return 0;
 }
 
+static inline bool is_spk(int id)
+{
+	if (id == 0x1308)
+		return true;
+
+	return false;
+}
```

Comment:
oh gosh this is terrible, sorry... We should really not encode part numbers this way. I think we really want to have information in ACPI tables telling which devices are part of the same group.

BTW this is how it was supported by our Windows friends